storage/innobase/handler/i_s.cc

```diff
@@ -6721,7 +6739,7 @@ static int i_s_innodb_tablespaces_fill_table(THD *thd, TABLE_LIST *tables,
     mtr_commit(&mtr);
     mutex_exit(&dict_sys->mutex);
 
-    if (ret && space != 0) {
+    if (ret) {
```

Comment:
Strictly speaking this is the fix, and i_s_get_filepath is refactoring only?

Reply:
My initial fix was only this :) and then it was crashing because of assert in fil_space_get_first_path().

  ut_a(space_id != TRX_SYS_SPACE);

The assert is valid. It says, you shouln't use this API for multi-file tablespaces. It is wrong to consider only first file when calculating size, block_size etc.

So the above refactoring avoids calls to fil_space_get_first_path() for system tablespace.

Reply:
Ah yes, now I see it

Reply:
Even better fix (future proof) would be to iterate all files of system tablespace and get the total size. We have to assume all files of system tablespace reside in one partition(or different partitions but of same block size).

But currently it is not seen necessary as Page IO compression is not enabled on system tablespace.